k2/csrc/fsa.h

```diff
@@ -12,6 +12,7 @@
 #include <utility>
 #include <vector>
 
+#include "glog/logging.h"
```

Comment:
glog should not be included in the header file.

Reply:
So how can you use `CHECK` in header file?

Reply:
You can put the implementation into `.cc` file.

If other projects that will use `fsa.h` and has its own `CHECK`, the compiler will
issue `macro redefinition` warnings.

Reply:
So we can never use glog in template class? that does not make sense.